util/file_checksum_helper.h

```diff
@@ -9,25 +9,29 @@
 #include "rocksdb/file_checksum.h"
 #include "rocksdb/status.h"
 #include "util/crc32c.h"
+#include "util/string_util.h"
 
 namespace rocksdb {
 
 // This is the class to generate the file checksum based on Crc32. It
 // will be used as the default checksum method for SST file checksum
 class FileChecksumFuncCrc32c : public FileChecksumFunc {
  public:
-  uint32_t Extend(uint32_t init_checksum, const char* data, size_t n) override {
+  std::string Extend(const std::string& init_checksum, const char* data,
+                     size_t n) override {
     assert(data != nullptr);
-    return crc32c::Extend(init_checksum, data, n);
+    uint32_t checksum_value = ParseUint32(init_checksum);
+    return ToString(crc32c::Extend(checksum_value, data, n));
```

Comment:
Let's not use `ToString()` which uses `std::to_string()`.  We should consider endianness and convert the result to byte array of 8. An example is https://github.com/facebook/rocksdb/blob/master/examples/multi_processes_example.cc#L74. We probably have something similar for uint32_t, but you may want to search the codebase.

Reply:
Just realized https://github.com/facebook/rocksdb/blob/master/port/port_posix.h#L99 can be used to detect endianness instead of writing a simple function.